source/geotiff_reader.h

```diff
@@ -133,19 +133,22 @@ namespace ryujin
       const auto &[di, dj] = apply_inverse_transformation(x, y);
 
       /*
-       * Use a simple bilinear interpolation:
+       * Use a simple bilinear interpolation and ensure we never go below the 0
+       * index due to round off error:
        */
 
-      const auto i_left = static_cast<unsigned int>(std::floor(di));
-      const auto i_right = static_cast<unsigned int>(std::ceil(di));
-      const auto j_left = static_cast<unsigned int>(std::floor(dj));
-      const auto j_right = static_cast<unsigned int>(std::ceil(dj));
+      const int min = 0;
+
+      const auto i_left = std::max(static_cast<int>(std::floor(di)), min);
+      const auto i_right = std::max(static_cast<int>(std::ceil(di)), min);
+      const auto j_left = std::max(static_cast<int>(std::floor(dj)), min);
+      const auto j_right = std::max(static_cast<int>(std::ceil(dj)), min);
```

Comment:
```suggestion
      const int i_left = std::max(static_cast<int>(std::floor(di)), 0);
      const int i_right = std::min(static_cast<int>(std::ceil(di)), raster_size_[0]);
      const int j_left = std::max(static_cast<int>(std::floor(dj)), 0);
      const int j_right = std::min(static_cast<int>(std::ceil(dj)), raster_size_[1]);
```

Reply:
Correction: We need to enforce min and max for all four to also handle the case that a coordinate is fully out of bounds.